src/lightning/pytorch/plugins/precision/xla.py

```diff
@@ -59,7 +59,8 @@ def __init__(self, precision: _PRECISION_INPUT = "32-true") -> None:
         else:
             self._desired_dtype = torch.float32
 
-    def optimizer_step(  # type: ignore[override]
+    @override  # type: ignore[override]
```

Comment:
shouldn't the type ignore annotation stay on the line below?

Reply:
I believe that would trigger a mypy error, but I will try and confirm.

Reply:
It doesn't throw an error (see commit [6fb730a](https://github.com/Lightning-AI/lightning/pull/19025/commits/6fb730ab7b725e0af3f29e5e351aef49d0cd3e1b)), so there appears to be some freedom with regards to the exact line of the ignore annotation. I thought I had put it on the top line in earlier PRs because an error was raised (e.g. https://github.com/Lightning-AI/lightning/blob/master/src/lightning/pytorch/strategies/xla.py#L73), but I must have remembered incorrectly. Where do you prefer the annotation?

Reply:
Since the ignore comment applies to the signature and not the decorator, it should be there in my opinion :)

Reply:
Just saw your commit fixing this - thanks!